app/views/unattended/provisioning_templates/host_init_config/host_init_config_default.erb

```diff
@@ -31,14 +48,7 @@ set -e
 <% end -%>
 
 # Call home to exit build mode
-<% built_https = foreman_url('built').start_with?('https') -%>
-<% if built_https -%>
-SSL_CA_CERT=$(mktemp)
-cat << EOF > $SSL_CA_CERT
-<%= foreman_server_ca_cert %>
-EOF
-<% end -%>
-curl --silent <%= '--cacert $SSL_CA_CERT' if built_https -%> -o /dev/null --noproxy \* '<%= foreman_url('built') %>'
+curl --silent --show-error <%= '--cacert $SSL_CA_CERT' if built_https -%> -o /dev/null --noproxy \* '<%= foreman_url('built') %>' || true
```

Comment:
Why `|| true`? I get that you may not want to trigger `exit_and_cancel_build`but if it fails, the built status is not set but it does echo "Successfully enrolled". Perhaps it should be `|| echo 'Failed to set built status'` so the user at least has some clue.

You then do get "failed" + "Success" as a user which may be confusing. A bit longer, but how about:
```bash
# Clear the error handler
trap - ERR

# Call home to exit build mode
curl --silent --show-error <%= '--cacert $SSL_CA_CERT' if built_https -%> -o /dev/null --noproxy \* '<%= foreman_url('built') %>
if [[ $? == 0 ]] ; then
  echo "Successfully enrolled host <%= @host.name %> with Foreman."
else
  echo "Successfully configured host <%= @host.name %> but failed to set built status in Foreman."
fi
```

Exact phrasing can be better, this is mostly about the structure.